share/shwap/row_namespace_data.go

```diff
@@ -122,6 +124,18 @@ func RowNamespaceDataFromProto(row *pb.RowNamespaceData) RowNamespaceData {
 	}
 }
 
+// ReadRowNamespaceData reads length-delimited protobuf representation of RowNamespaceData and
+// returns it with total number of bytes written.
+func ReadRowNamespaceData(reader io.Reader) (RowNamespaceData, int, error) {
```

Comment:
Made it a func, so that:
* NamespaceData.ReadFrom remains atomic
* Goland is not going crazy about pointer vs value methods(optional)

Reply:
I find shwap pkg to be very heavy in terms of amount of function already. can be also RowNamespaceData.ReadFrom(). It will hide this func from pkg level func surface to method associated with type. 
Goland linting is not a big deal